src/decoder.c

```diff
@@ -279,7 +279,7 @@ int _enter_container(nanocbor_value_t *it, nanocbor_value_t *container,
     container->end = it->end;
     container->remaining = 0;
 
-    if (_value_match_exact(it, type | NANOCBOR_VALUE_MASK) == 1) {
+    if (_value_match_exact(it, (type << NANOCBOR_TYPE_OFFSET | NANOCBOR_SIZE_INDEFINITE)) == 1) {
```

Comment:
```suggestion
    if (_value_match_exact(it, ((type << NANOCBOR_TYPE_OFFSET) | NANOCBOR_SIZE_INDEFINITE)) == 1) {
```

Does that help? (it makes it at least easier to parse for humans)

Reply:
lint is still not satisfied by this change

Reply:
hmm I don't know either :worried:  - maybe try casting it to `(unsigned)`

Reply:
Now lint is happy. This is so weird. The `type` variable was already `uint8_t`. :confused:

Reply:
Sure, it's up to @bergzand to merge this though